src/transformers/utils/hub.py

```diff
@@ -221,7 +221,8 @@ def extract_commit_hash(resolved_file: Optional[str], commit_hash: Optional[str]
     """
     if resolved_file is None or commit_hash is not None:
         return commit_hash
-
+    if isinstance(resolved_file, str) and sys.platform == "win32":
```

Comment:
```suggestion
    if isinstance(resolved_file, str):
```
No need for the test now, as `Path` is cross-library :-)